pkg/sorter/pebble/db.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pebble
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/cockroachdb/pebble/bloom"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/pkg/config"
+	"github.com/pingcap/tiflow/pkg/sorter"
+	"github.com/pingcap/tiflow/pkg/sorter/pebble/encoding"
+	"go.uber.org/zap"
+)
+
+const (
+	minTableCRTsLabel      string = "minCRTs"
+	maxTableCRTsLabel      string = "maxCRTs"
+	tableCRTsCollectorName string = "table-crts-collector"
+)
+
+type tableCRTsCollector struct {
+	minTs uint64
+	maxTs uint64
+}
+
+func (t *tableCRTsCollector) Add(key pebble.InternalKey, value []byte) error {
+	crts := encoding.DecodeCRTs(key.UserKey)
+	if crts > t.maxTs {
+		t.maxTs = crts
+	}
+	if crts < t.minTs {
+		t.minTs = crts
+	}
+	return nil
+}
+
+func (t *tableCRTsCollector) Finish(userProps map[string]string) error {
+	userProps[minTableCRTsLabel] = fmt.Sprintf("%d", t.minTs)
+	userProps[maxTableCRTsLabel] = fmt.Sprintf("%d", t.maxTs)
+	return nil
+}
+
+func (t *tableCRTsCollector) Name() string {
+	return tableCRTsCollectorName
+}
+
+// NOTE: both lowerBound and upperBound are included.
+func iterTable(
```

Comment:
why are both bound included instead of one bound?

Reply:
Generally `upperBound` comes from `resolvedTs` or `barrierTs`, which are included.
`lowerBound` can be included or excluded, I think both are OK.